frontend/packages/dev-console/src/components/import/import-submit-utils.ts

```diff
@@ -823,3 +825,74 @@ export const handleRedirect = async (
   const redirectURL = (await perspectiveData.properties.importRedirectURL())(project);
   history.push(redirectURL);
 };
+
+export const getTelemetryImport = (values: GitImportFormData) => {
+  if (!values) return {};
+  // get devfile data if exists
+  let devfileJson;
+  if (values.devfile?.devfileContent) {
+    devfileJson = safeYAMLToJS(values.devfile.devfileContent);
+  }
+  const applicationType =
+    (!values.application?.selectedKey && values.application?.name) ||
+    values.application?.selectedKey === values.application?.name
+      ? 'default'
+      : values.application?.selectedKey === CREATE_APPLICATION_KEY && !!values.application?.name
+      ? 'custom'
+      : 'none';
+  const resourceType =
+    values.resources === Resources.Kubernetes
+      ? 'Deployment'
+      : values.resources === Resources.OpenShift
+      ? 'DeploymentConfig'
+      : values.resources === Resources.KnativeService && 'Knative Service';
+  const selStrategy = values.import?.selectedStrategy?.name;
+  return {
+    useRecommended: values.image?.recommended === values.image?.selected,
+    recommendedStrategy: values.import?.recommendedStrategy?.name,
+    recommendedBuilderImage: values.image?.recommended,
+    strategy: selStrategy,
+    builderImage: values.image?.selected,
+
+    devFileLanguage: devfileJson?.metadata?.projectType || '',
+    devFileProjectType: devfileJson?.metadata?.language || '',
+
+    application: applicationType,
+
+    resource: resourceType,
+
+    targetPortChanged: !!values.route?.unknownTargetPort,
+
+    // currently run command is being used only for Node app
+    useRunCommand: !!values.image?.imageEnv?.NPM_RUN,
+
+    useAdvancedOptionsGit: values.git.dir !== '/' || !!values.git.ref || !!values.git.secret,
+
+    useAdvancedOptionsBuild:
+      values.build?.env.length > 0 ||
+      !values.build?.triggers?.config ||
+      !values.build?.triggers?.image ||
+      !values.build?.triggers?.webhook,
+    useAdvancedOptionsDeployment:
+      selStrategy !== 'Devfile' &&
+      (values.deployment?.env.length > 0 || !values.deployment?.triggers?.image),
+    useAdvancedOptionsRoute:
+      !!values.route?.hostname ||
+      !!values.route?.path ||
+      !values.route?.secure ||
```

Comment:
```suggestion
      !!values.route?.secure ||
```

Reply:
same here am not trying to typecase to boolean here, secure by default is true so if it's false i.e it's changed

Reply:
Same as above.